aten/src/ATen/core/ivalue.cpp

```diff
@@ -533,16 +533,31 @@ IValueComparator getLessThanComparator(const IValue& v) {
       std::vector<IValueComparator> elements_lts;
       elements_lts.reserve(n);
       for (size_t i = 0; i < n; ++i) {
-        elements_lts.push_back(getLessThanComparator(elements[i]));
+        if (elements[i].isNone()) {
+          elements_lts.push_back(nullptr);
+        } else {
+          auto lt_func = getLessThanComparatorWithNone(elements[i], why_not);
+          if (lt_func == nullptr)
+            return nullptr;
+          elements_lts.push_back(lt_func);
+        }
       }
 
       return [elements_lts=std::move(elements_lts), n](const IValue& a, const IValue& b) {
         const auto& a_elements = a.toTuple()->elements();
         const auto& b_elements = b.toTuple()->elements();
 
         for (size_t i = 0; i < n; ++i) {
-          if (elements_lts[i](a_elements[i], b_elements[i])) {
-            return true;
+          if (elements_lts[i] != nullptr) {
+            // a_elements[i] is not None, use the comparator.
```

Comment:
Agreed. In addition, I think that sampling one list element and using it to build the comparator function cannot work in the presence of `Optional[T]`. For example, in the list `[(3, None), (None, 4)]`, there is no one element that you can sample in order to build the right comparator. 

Since the list that is to be sorted is traversed in `checkAndGetComparator` anyway, it might make sense to try and infer the type of the list elements by scanning each one and passing that information into this function so that the right comparator can be built.